### ISSUE TITLE
Proguard Rules (OkHttp 4.7.2)

### ISSUE THREAD

**Issue by rewgoes** [square/okhttp#6258]
Hi,

I am getting a few warnings when building with DexGuard.

Gradle: 5.6.4
Android Gradle Plugin: 3.6.3
OkHttp: 4.7.2
compile_sdk = 30
build_tools = "30.0.2"

Current Proguard rules for OkHttp:
```
# ========= START OkHttp =========
# JSR 305 annotations are for embedding nullability information.
-dontwarn javax.annotation.**

# A resource is loaded with a relative path so the package of this class must be preserved.
-keepnames class okhttp3.internal.publicsuffix.PublicSuffixDatabase

# Animal Sniffer compileOnly dependency to ensure APIs are compatible with older versions of Java.
-dontwarn org.codehaus.mojo.animal_sniffer.*

# OkHttp platform used only on JVM and when Conscrypt dependency is available.
-dontwarn okhttp3.internal.platform.ConscryptPlatform
# ========= END OkHttp =========
```

```
Warning: okhttp3.internal.platform.ConscryptPlatform$platformTrustManager$2: can't find superclass or interface org.conscrypt.ConscryptHostnameVerifier
Warning: okhttp3.internal.io.FileSystem$Companion$SYSTEM$1: can't find referenced class okhttp3.internal.io.FileSystem$DefaultImpls
Warning: okhttp3.internal.io.FileSystem$Companion$SYSTEM$1: can't find referenced class okhttp3.internal.io.FileSystem$DefaultImpls
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.BCSSLParameters
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.BCSSLParameters
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.provider.BouncyCastleJsseProvider
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.provider.BouncyCastleJsseProvider
Warning: okhttp3.internal.platform.BouncyCastlePlatform: can't find referenced class org.bouncycastle.jsse.BCSSLParameters
Warning: okhttp3.internal.platform.ConscryptPlatform$Companion: can't find referenced class org.conscrypt.Conscrypt
Warning: okhttp3.internal.platform.ConscryptPlatform$Companion: can't find referenced class org.conscrypt.Conscrypt
Warning: okhttp3.internal.platform.ConscryptPlatform$Companion: can't find referenced class org.conscrypt.Conscrypt$Version
Warning: okhttp3.internal.platform.ConscryptPlatform$Companion: can't find referenced class org.conscrypt.Conscrypt$Version
Warning: okhttp3.internal.platform.ConscryptPlatform$Companion: can't find referenced class org.conscrypt.Conscrypt$Version
Warning: okhttp3.internal.platform.ConscryptPlatform$Companion: can't find referenced class org.conscrypt.Conscrypt$Version
Warning: okhttp3.internal.platform.ConscryptPlatform$Companion: can't find referenced class org.conscrypt.Conscrypt$Version
Warning: okhttp3.internal.platform.ConscryptPlatform$platformTrustManager$2: can't find referenced class org.conscrypt.ConscryptHostnameVerifier
Warning: okhttp3.internal.platform.OpenJSSEPlatform: can't find referenced class org.openjsse.javax.net.ssl.SSLSocket
Warning: okhttp3.internal.platform.OpenJSSEPlatform: can't find referenced class org.openjsse.javax.net.ssl.SSLSocket
Warning: okhttp3.internal.platform.OpenJSSEPlatform: can't find referenced class org.openjsse.javax.net.ssl.SSLParameters
Warning: okhttp3.internal.platform.OpenJSSEPlatform: can't find referenced class org.openjsse.javax.net.ssl.SSLParameters
Warning: okhttp3.internal.platform.OpenJSSEPlatform: can't find referenced class org.openjsse.javax.net.ssl.SSLSocket
Warning: okhttp3.internal.platform.OpenJSSEPlatform: can't find referenced class org.openjsse.javax.net.ssl.SSLSocket
Warning: okhttp3.internal.platform.OpenJSSEPlatform: can't find referenced class org.openjsse.net.ssl.OpenJSSE
Warning: okhttp3.internal.platform.OpenJSSEPlatform: can't find referenced class org.openjsse.net.ssl.OpenJSSE
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLParameters
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLParameters
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLParameters
Warning: okhttp3.internal.platform.android.BouncyCastleSocketAdapter: can't find referenced class org.bouncycastle.jsse.BCSSLSocket
Warning: okhttp3.internal.platform.android.ConscryptSocketAdapter: can't find referenced class org.conscrypt.Conscrypt
Warning: okhttp3.internal.platform.android.ConscryptSocketAdapter: can't find referenced class org.conscrypt.Conscrypt
Warning: okhttp3.internal.platform.android.ConscryptSocketAdapter: can't find referenced class org.conscrypt.Conscrypt
Warning: okhttp3.internal.platform.android.ConscryptSocketAdapter: can't find referenced class org.conscrypt.Conscrypt
Warning: okhttp3.internal.platform.android.ConscryptSocketAdapter: can't find referenced class org.conscrypt.Conscrypt
Warning: okhttp3.logging.HttpLoggingInterceptor$Logger$Companion$DEFAULT$1: can't find referenced class okhttp3.logging.HttpLoggingInterceptor$Logger$DefaultImpls
Warning: okhttp3.logging.HttpLoggingInterceptor$Logger$Companion$DEFAULT$1: can't find referenced class okhttp3.logging.HttpLoggingInterceptor$Logger$DefaultImpls
```

Are these warnings safe to ignore (dontwarn)?

The app seems to be working as expected after obfuscation, but I don't want to have any surprises in the future cause by the obfuscation.

**Comment by hungntv** [comment]
I can't reproduce the issue and I don't know how to do it.

You can try to add this: `implementation 'org.conscrypt:conscrypt-android:2.2.1'`

**Comment by WonderCsabo** [comment]
When running with Android Gradle Plugin 7.0, R8 prints this warning:

```
WARNING:Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in app/build/outputs/mapping/devRelease/missing_rules.txt.

WARNING:R8: Missing class org.bouncycastle.jsse.BCSSLParameters (referenced from: void okhttp3.internal.platform.BouncyCastlePlatform.configureTlsExtensions(javax.net.ssl.SSLSocket, java.lang.String, java.util.List) and 1 other context)
Missing class org.bouncycastle.jsse.BCSSLSocket (referenced from: void okhttp3.internal.platform.BouncyCastlePlatform.configureTlsExtensions(javax.net.ssl.SSLSocket, java.lang.String, java.util.List) and 5 other contexts)
Missing class org.bouncycastle.jsse.provider.BouncyCastleJsseProvider (referenced from: void okhttp3.internal.platform.BouncyCastlePlatform.<init>())
Missing class org.conscrypt.Conscrypt$Version (referenced from: boolean okhttp3.internal.platform.ConscryptPlatform$Companion.atLeastVersion(int, int, int))
Missing class org.conscrypt.Conscrypt (referenced from: boolean okhttp3.internal.platform.ConscryptPlatform$Companion.atLeastVersion(int, int, int) and 4 other contexts)
Missing class org.conscrypt.ConscryptHostnameVerifier (referenced from: okhttp3.internal.platform.ConscryptPlatform$DisabledHostnameVerifier)
Missing class org.openjsse.javax.net.ssl.SSLParameters (referenced from: void okhttp3.internal.platform.OpenJSSEPlatform.configureTlsExtensions(javax.net.ssl.SSLSocket, java.lang.String, java.util.List))
Missing class org.openjsse.javax.net.ssl.SSLSocket (referenced from: void okhttp3.internal.platform.OpenJSSEPlatform.configureTlsExtensions(javax.net.ssl.SSLSocket, java.lang.String, java.util.List) and 1 other context)
Missing class org.openjsse.net.ssl.OpenJSSE (referenced from: void okhttp3.internal.platform.OpenJSSEPlatform.<init>())
```

I think okhttp should ship these rules.

**Comment by itsandreramon** [comment]
+1

**Comment by tajchert** [comment]
I think it should be safe to use `dontwarn `as those classes are not used on Android anyway.

```
-dontwarn org.bouncycastle.jsse.BCSSLSocket
-dontwarn org.bouncycastle.jsse.BCSSLParameters
-dontwarn org.bouncycastle.jsse.provider.BouncyCastleJsseProvider
-dontwarn org.conscrypt.*
-dontwarn org.openjsse.javax.net.ssl.SSLParameters
-dontwarn org.openjsse.javax.net.ssl.SSLSocket
-dontwarn org.openjsse.net.ssl.OpenJSSE
```

**Comment by mobilekosmos** [comment]
"Aare not used on Android anyway", hm? why is that code then there?

**Comment by WonderCsabo** [comment]
OkHttp supports [Java platforms](https://github.com/square/okhttp/tree/master/okhttp/src/main/kotlin/okhttp3/internal/platform) other than Android, as well.

**Comment by yschimke** [comment]
Fixed in 5.0 Alpha3 https://github.com/square/okhttp/pull/6792, happy to have any one confirm the fix is working ahead of 5.0

**Comment by omishah** [comment]
> Fixed in 5.0 Alpha3 #6792, happy to have any one confirm the fix is working ahead of 5.0

Yes 🥂

**Comment by yuriidevua** [comment]
4.11.0 fix
